### PR TITLE
delete shader works

### PR DESCRIPTION
**Pull request by brucelane**

### CODE REVIEW

File: spec/index.js [hunk 1]
```diff
@@ -39,5 +39,27 @@ describe('Controllers', function() {
                     done();
                 });
         });
+
+        it('List shaders successfully', function(done) {
+            saveStub
```
Comment (jarpi) [inline spec/index.js]: In order to functionalities, create a new 'listStub', like the saveStub one: 
listStub = sinon.stub(shaderModel.prototype, 'findOne');

File: spec/index.js [hunk 2]
```diff
@@ -39,5 +39,27 @@ describe('Controllers', function() {
...
+            .yields(null, {});
+            shaderController.getShaderList()
+                .then(function(value){
+                done();
+            });
+        });
+
+        it('Delete a shader fails', function(done) {
+            saveStub
```
Comment (jarpi) [inline spec/index.js]: Same for delete

File: controllers/test.js [hunk 3]
```diff
@@ -33,17 +33,15 @@ function listShaders() {
...
-            console.log(shaderId);
-            console.log(shaderToDelete);
             if (err) {
                 console.log(err);
                 return reject(err);
             }
-            console.log("ok deleted");
-            return resolve(shaderToDelete);
-        
+            shaderToDelete.remove();
```
Comment (jarpi) [inline controllers/test.js]: Isn't there a callback function to handle errors and documents deleted?

File: spec/index.js (hunk 1 above)
Comment (brucelane) [inline spec/index.js]: going to try

File: spec/index.js (hunk 2 above)
Comment (brucelane) [inline spec/index.js]: going to try

File: controllers/test.js (hunk 3 above)
Comment (brucelane) [inline controllers/test.js]: I don't know...

File: controllers/test.js [hunk 4]
```diff
@@ -33,17 +33,15 @@ function listShaders() {
 }
 function removeShader(shaderId) {
     return new Promise(function(resolve, reject) {
-       Shader.findOneAndRemove({id: shaderId}).exec( 
+       Shader.findOneAndRemove({_id: shaderId}).exec(
```
Comment (jarpi) [inline controllers/test.js]: Shader.findOneAndRemove({_id: shaderId}, null, function(error, doc, result) {
  // error: any errors that occurred
  // doc: the document before updates are applied if `new: false`, or after updates if `new = true`
  // result: [raw result from the MongoDB driver](http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#findAndModify)
if (err) reject... 
resolve(doc)...
}
)

Comment (brucelane) [inline controllers/test.js]: I did that already !

Comment (jarpi) [inline controllers/test.js]: Ok, in order to stub find and so on, works a little different. We have to stub the prototype function inherited when mongoose creates the instance of the model: 
At the top: 
var mongoose = require('mongoose'); 

In before: 
listStub = sinon.stub(mongoose.Model, 'find');
deleteStub = sinon.stub(mongoose.Model, 'findOneAndRemove');

**Pull request merged**